apps/web/src/components/ConnectWalletButton/ConnectWalletButton.tsx

```diff
@@ -1,10 +1,12 @@
 import { ethers } from "ethers";
 import React, { useEffect, useState } from "react";
 import Button from "@mui/material/Button";
+import { currentWalletState } from "../../state";
+import { useRecoilState } from "recoil";
 
 const ConnectWalletButton: React.FC = () => {
+  const [currentWallet, setCurrentWallet] = useRecoilState(currentWalletState);
   const [haveMetamask, sethaveMetamask] = useState(true);
-  const [accountAddress, setAccountAddress] = useState("");
   const [accountBalance, setAccountBalance] = useState("");
   const [isConnected, setIsConnected] = useState(false);
```

Comment:
let's refactor all these state values into recoil, instead of creating that one state atom for everything.
wdyt? anyway even if you just have one atom for everything, lets refactor all these state declarations out and use recoil.

Reply:
I keep the haveMetamask the same because it's just a state to check if the browser has metamask installed (we currently use it for UI change only and not for the user info). For the rest, I have refactored them all to userDetailsState! ^^